src/react-query/impl/infinite/hooks.ts

```diff
@@ -11,10 +14,10 @@ export const useInfiniteQueryData = <TDataSource extends AnyInfiniteQueryDataSou
     context: DataSourceContext<TDataSource>,
     dataSource: TDataSource,
     params: DataSourceParams<TDataSource>,
-    options?: DataSourceOptions<TDataSource>,
-) => {
+    options?: Partial<DataSourceOptions<TDataSource>>,
```

Comment:
Can you move partial type to DataSourceOptions?

Reply:
I believe that having origin options can be useful